plugins/module_utils/redfish_utils.py

```diff
@@ -65,11 +70,32 @@ def get_request(self, uri):
         req_headers = dict(GET_HEADERS)
         username, password, basic_auth = self._auth_params(req_headers)
         try:
-            resp = open_url(uri, method="GET", headers=req_headers,
-                            url_username=username, url_password=password,
-                            force_basic_auth=basic_auth, validate_certs=False,
+            
+            if os.path.isfile(HOME + "/sessionfile" + self.root_uri[8:] + ".txt"):
```

Comment:
What exactly are the contents of this file? How is a user supposed to fill this out?

Reply:
And what's the significance of `self.root_uri[8:]` in the file name?

Reply:
self.root_uri[8:] is the session id. Txt will be created as sessionfile<sessionid>.txt keeping in mind the uniqueness of this file.  In other words, if there is another connection to different server, then the file will unique.

Reply:
This seems to be a very specific convention that's not followed by everyone. Is it actually documented somewhere? Who is following it besides this implementation?

Reply:
@felixfontein ,  In iLORest tool, we follow this to cache session id in the %appdata%/text file directory.  As you are aware, session id caching can be done in various implementations.  And we have adopted this way for now.  We can always improve on this in future iterations.  Also, this code does not break existing functionality. On that pretext, can we able to approve and merge it for now and look for improvements later?

Reply:
@mraineri , we have completed all the changes and tested it out as well.  Any enhancements we can take it in future

Reply:
@mraineri , any idea how much more time it takes because we plan to release our ansible collection very soon to automationhub/ansiblegalaxy.  And the ansible collection depends on these upstream changes.
Thanks
Rajeev

Reply:
@rajeevarakkal you probably won't be able to publish your collection to AutomationHub if you depend on community.general, since c.g is not available there (as it is not certified).

> any idea how much more time it takes

That depends on how long it takes to bring this PR into a form that can be merged. I really suggest to remove all that file-specific code from this PR, and instead add a mechanism which allows your code to provide the token (read from the file) in another way (as a new parameter or something like that).

Reply:
ok Mike, We will see what we can do.

Reply:
We have removed the file implementation and added the sessionid as part of the header. Check the lastest commit for these changes.

Reply:
@mraineri , Please check out our latest changes and please approve.